index.js

```diff
@@ -199,40 +196,42 @@ function filesToStream(filesPromise, options) {
 }
 
 /**
- * @param {Array<{entity: BemEntityName, level: String, tech: String, path: String}>} introspection - unordered file-entities list
- * @param {String[]} levels - ordered levels' paths list
- * @param {Tenorok[]} decl - resolved and ordered declaration
+ * @param {Object} opts - Options for harvester
+ * @param {Array<{entity: BemEntityName, level: String, tech: String, path: String}>} opts.introspection - unordered file-entities list
+ * @param {String[]} opts.levels - ordered levels' paths list
+ * @param {Object<String, String[]>} [opts.techMap] - deps techs to file techs mapper
+ * @param {BemCell[]} opts.decl - resolved and ordered declaration
  * @returns {Array<{entity: BemEntityName, level: String, tech: String, path: String}>} - resulting ordered file-entities list
  */
-function harvest(introspection, levels, decl/*: Array<{entity, tech}>*/) {
-    const hash = fileEntity => `${fileEntity.entity.id}.${fileEntity.tech}`;
-    const declIndex = _buildIndex(decl, hash);
-
-    const entityInIndex = file => declIndex[hash(file)] !== undefined;
-    return introspection
-        .filter(entityInIndex)
-        .filter(file => levels.indexOf(file.level) !== -1)
-        .sort((f1, f2) => hash(f1) === hash(f2)
-            ? levels.indexOf(f1.level) - levels.indexOf(f2.level)
-            : declIndex[hash(f1)] - declIndex[hash(f2)]);
+function harvest(opts) {
+    const hash = cell => `${cell.entity.id}.${cell.tech}`;
+    const declIndex = _buildIndex(opts.decl, hash);
+
+    const fileTechToDep = Object.keys(opts.techMap || {}).reduce((res, depTech) =>
```

Comment:
dependency tech to file system tech.

`{ block: 'button', tech: 'react.js' }` by default should look for `**/button/button.react.js` (roughly).
But with techMap you can force it to look for, let's say, `**/button/button.component.js` or some else.

It's definitely not extension.